### PR TITLE
ci: use pypi trusted publisher

### PR DESCRIPTION
**Pull request by Jonas1312**
I don't have access to our pypi account

@PierreLeveau you have to follow this tutorial: https://docs.pypi.org/trusted-publishers/adding-a-publisher/

on test pypi account too

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -35,13 +37,14 @@ jobs:
...
-        run: |
-          python -m twine upload --repository-url https://test.pypi.org/legacy/ dist/*
-          python -m twine upload dist/*
+      - name: Publish to test PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
+
+      - name: Publish to PyPI
+        if: startsWith(github.ref, 'refs/tags')
```
Comment (PierreLeveau) [inline .github/workflows/publish.yml:46]: why is this added?

Comment (Jonas1312) [reply .github/workflows/publish.yml:46]: to check that the commit has a tag: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#publishing-the-distribution-to-pypi-and-testpypi
